joblib/func_inspect.py

```diff
@@ -169,7 +170,11 @@ def getfullargspec(func):
     try:
         return inspect.getfullargspec(func)
     except AttributeError:
-        arg_spec = inspect.getargspec(func)
+        if 3 > sys.version_info[0]:
+            arg_spec = inspect.getargspec(func)
```

Comment:
I may be missing something but I don't get your change at all. If you have Python3 inspect.getfullargspec is used and you never find yourself in this except block ...